src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -96,17 +99,17 @@ internal override bool IsIUnknownConstant
 
         internal override bool IsCallerLineNumber
         {
-            get { throw ExceptionUtilities.Unreachable; }
+            get { return false; }
         }
 
         internal override bool IsCallerFilePath
         {
-            get { throw ExceptionUtilities.Unreachable; }
+            get { return false; }
         }
 
         internal override bool IsCallerMemberName
         {
-            get { throw ExceptionUtilities.Unreachable; }
+            get { return false; }
         }
```

Comment:
Are these reachable now?

Reply:
Yes, see the local function for binding default parameters in the invocation binder here: https://github.com/dotnet/roslyn/blob/622fc4469a09f7511bebefc2462955b5d889de87/src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs#L1359. It accesses those properties, and we may now be passing in `SynthesizedParameterSymbol`  as the parameter

Reply:
@adamperlin I see, but it seems there is no test coverage for caller info, and returning false doesn't feel correct to me (unless I'm missing something). We need some tests similar to this case anyway:

```csharp
Action<string> x = ([CallerMemberName] x = "") => Console.WriteLine(x);
x();
```

Whether this is something to address in this PR or in a follow-up is the compiler's team decision. I think a PROTOTYPE comment might be sufficient for now.

Reply:
Let's add these tests now.

Reply:
Added test coverage for the  `Caller...` attributes. The attributes are currently ignored

Reply:
I have added a PROTOTYPE comment to address this issue in LDM